telegram-translation-shortcuts.user.js

```diff
@@ -66,20 +67,76 @@ function scrollItems (down) {
   }
 }
 
+/**
+ * A div with className 'key-add-suggestion-wrap' contains the input-form with Submit and Cancel buttons to add translation/suggestion.
+ * If the input wrapper is collapsed (closed), the className will change to 'key-add-suggestion-wrap collapsed'
+ *
+ * @function Checks if input-form (key-add-suggestion-wrap) is open or closed
+ * @returns True if Open
+ * @returns False if Closed
+ */
+function inputTranslationOpen () {
+  return (document.getElementsByClassName('key-add-suggestion-wrap')[0].className === 'key-add-suggestion-wrap')
+}
+
 /**
  * Clicks the 'Add Translation' button
  */
 function addTranslation () {
-  document.getElementsByClassName('key-add-suggestion-header').item(0).click()
+  // Prevent toggle behaviour when clicking
+  if (inputTranslationOpen()) { // Focus back to input box (don't lose what was typed)
+    document.getElementsByClassName('tr-form-control').item(0).focus()
+  } else { // Open the input-wrapper
+    document.getElementsByClassName('key-add-suggestion-header').item(0).click()
+  }
 }
 
 /**
- * Clicks the edit icon
+ * Clicks the edit icon.
+ * Useful when a suggestion exists but no translation is applied.
  */
 function editTranslation () {
-  document.getElementsByClassName('ibtn key-suggestion-edit').item(0).click()
+  // Don't click if already typing something
+  if (inputTranslationOpen()) { // Focus back to input box
+    document.getElementsByClassName('tr-form-control').item(0).focus()
+  } else {
+    document.getElementsByClassName('ibtn key-suggestion-edit').item(0).click()
+  }
+}
+
+/**
+ * Clicks the cancel button
+ */
+function cancelTranslation () {
+  // Prevent visual glitch: Don't cancel if input-wrapper is already closed, but focus somehow remains in the 'form-control' element
```

Comment:
I checked and found out that, there are at least 2 other elements with 'form-control' in their className. That includes 'search' and 'login'